apps/concierge_site/lib/helpers/mjml.ex

```diff
@@ -0,0 +1,93 @@
+defmodule ConciergeSite.MJML do
+  @moduledoc """
+  Utilities to make working with our MJML templates similar to working with
+  typical EEx templates.
+  """
+
+  @template_dir Application.compile_env!(:concierge_site, :mjml_template_dir)
+
+  @spec template_path(Path.t()) :: Path.t()
+  def template_path(name) do
+    Path.join(@template_dir, name)
+  end
+
+  # NOTE(Nick): This escape/unescape is necessary since the underlying MJML
+  # library we rely on doesn't support arbitrary junk properly inside
+  # <mjml-raw> tags. See https://github.com/jdrouet/mrml/issues/252 for an
+  # issue tracking this.
+
+  defp escape_eex(contents) do
+    contents = Regex.replace(~r/<%/, contents, "EEX_LHS_ESCAPED")
+
+    Regex.replace(~r/%>/, contents, "EEX_RHS_ESCAPED")
+  end
+
+  defp unescape_eex(contents) do
+    contents = Regex.replace(~r/EEX_LHS_ESCAPED/, contents, "<%")
+
+    Regex.replace(~r/EEX_RHS_ESCAPED/, contents, "%>")
+  end
+
+  @spec to_html_with_eex(String.t()) :: {:ok, String.t()} | {:error, any()}
+  def to_html_with_eex(contents) do
+    case Mjml.to_html(escape_eex(contents)) do
+      {:ok, contents} -> {:ok, unescape_eex(contents)}
+      e -> e
+    end
+  end
+
+  @spec html_from_template(Path.t()) :: {:ok, String.t()} | {:error, any()}
+  def html_from_template(name) do
```

Comment:
question: Could this function be private? (Same with, e.g., `to_html_with_eex`.)